pkg/controller/installation/core_controller.go

```diff
@@ -233,15 +235,29 @@ type ReconcileInstallation struct {
 	amazonCRDExists      bool
 }
 
-// GetInstallation returns the default installation instance with defaults populated.
+// GetInstallation returns the current installation resource.
 func GetInstallation(ctx context.Context, client client.Client, provider operator.Provider) (*operator.Installation, error) {
+	// TODO: remove unused provider
```

Comment:
I could use some assistance getting rid of this TODO.

Most of the other controllers are passed and store a provider at initilization just so that they can pass it to this function. Now that this function does not need to be passed a provider, it seems unnecessary for those other functions to pass the provider in at all when they can just pull it from the Installation resource. So I wonder if we can remove the provider from those structs. But I'm a bit concerned about making that big change. So I might prefer we do that after? Or delay this PR a bit to test out and implement that change.

Reply:
Yeah, I get not wanting to hold this up, but I sorta feel like it should be done as part of this PR, which is already a pretty big change.